runtime/sema/check_interface_declaration.go

```diff
@@ -557,6 +557,69 @@ func (checker *Checker) declareEntitlementMappingType(declaration *ast.Entitleme
 	return entitlementMapType
 }
 
+// Recursively resolve the include statements of an entitlement mapping declaration, walking the "hierarchy" defined in this file
+// Uses the sync primitive stored in `resolveInclusions` to ensure each map type's includes are computed only once.
+// This assumes that any includes coming from imported files are necessarily already completely resolved, since that imported file
+// must necessarily already have been fully checked. Additionally, because import cycles are not allowed in Cadence, we only
+// need to check for map-include cycles within the currently-checked file
+func (checker *Checker) resolveEntitlementMappingInclusions(
+	mapType *EntitlementMapType,
+	declaration *ast.EntitlementMappingDeclaration,
+	visitedMaps map[*EntitlementMapType]struct{},
+) {
+	mapType.resolveInclusions.Do(func() {
```

Comment:
Maybe make this a method on `EntitlementMapType`, it is a bit odd that `Checker` reaches into `EntitlementMapType` and updates its internal state (e.g. uses `resolveInclusions`)